### PR TITLE
Add KMS VPC endpoint

### PR DESCRIPTION
**Pull request by jonathanbeber**
This commit creates the logic required to add a KMS endpoint to each VPC
under the config.

### CODE REVIEW

File: sevenseconds/config/vpc.py [hunk 1]
```diff
@@ -622,6 +623,100 @@ def create_vpc_endpoints(account: AccountData, vpc: object, region: str):
...
+                sg_name = 'KMS VPC Endpoint'
+                sg_desc = 'Allow access to the KMS VPC endpoint'
+                sg = get_sg(sg_name, sg_desc, vpc.security_groups.all())
+                if not sg:
+                    sg = vpc.create_security_group(
+                        GroupName=sg_name,
+                        Description=sg_desc,
+                    )
+                    time.sleep(2)
+                    sg.create_tags(Tags=[{'Key': 'Name', 'Value': sg_name}])
```
Comment (aermakov-zalando) [inline sevenseconds/config/vpc.py]: `InfrastructureComponent: true` to prevent the users from breaking it?

File: sevenseconds/config/vpc.py [hunk 2]
```diff
@@ -622,6 +623,100 @@ def create_vpc_endpoints(account: AccountData, vpc: object, region: str):
...
+            with ActionOnExit('Checking VPC Endpoint {}..'.format(service_name)) as act:
+                sg_name = 'KMS VPC Endpoint'
+                sg_desc = 'Allow access to the KMS VPC endpoint'
+                sg = get_sg(sg_name, sg_desc, vpc.security_groups.all())
+                if not sg:
+                    sg = vpc.create_security_group(
+                        GroupName=sg_name,
+                        Description=sg_desc,
+                    )
+                    time.sleep(2)
```
Comment (aermakov-zalando) [inline sevenseconds/config/vpc.py]: Why?

File: sevenseconds/config/vpc.py [hunk 3]
```diff
@@ -622,6 +623,100 @@ def create_vpc_endpoints(account: AccountData, vpc: object, region: str):
                             ).encode('utf-8')).hexdigest()
                     }
                     act.warning('missing, make create: {}'.format(ec2c.create_vpc_endpoint(**options)))
+        elif service_name.endswith('.kms'):
+            with ActionOnExit('Checking VPC Endpoint {}..'.format(service_name)) as act:
```
Comment (aermakov-zalando) [inline sevenseconds/config/vpc.py]: Move this to a separate function maybe?

File: sevenseconds/config/vpc.py [hunk 4]
```diff
@@ -781,3 +876,20 @@ def cleanup_vpc(account: AccountData, region: str):
...
+            return sg
+
+
+def allow_https_all(permissions: list) -> bool:
+    for permission in permissions:
+        if (
+            permission['IpProtocol'] == 'tcp'
+            and permission['FromPort'] == 443
+            and permission['ToPort'] == 443
+            and {'CidrIp': '0.0.0.0/0'} in permission['IpRanges']
```
Comment (aermakov-zalando) [inline sevenseconds/config/vpc.py]: I'd still use the VPC IP address here, considering weird edge cases like peered VPCs and VPNs.

File: sevenseconds/config/vpc.py (hunk 2 above)
Comment (jonathanbeber) [reply sevenseconds/config/vpc.py]: https://github.com/zalando-stups/sevenseconds/blob/3976d8be790b5f4a6a7e5a517844eeca3835b5d1/sevenseconds/config/vpc.py#L295-L296
not sure if still relevant

File: sevenseconds/config/vpc.py [hunk 5]
```diff
@@ -622,6 +623,100 @@ def create_vpc_endpoints(account: AccountData, vpc: object, region: str):
...
+                                )
+                            if subnets.difference(subnet_in_endpoint):
+                                options['AddSubnetIds'] = list(
+                                    subnets.difference(subnet_in_endpoint)
+                                )
+                            act.warning(
+                                'mismatch ({} vs. {}), make update: {}'.format(
+                                    subnet_in_endpoint,
+                                    subnets,
+                                    ec2c.modify_vpc_endpoint(**options),
```
Comment (aermakov-zalando) [inline sevenseconds/config/vpc.py]: I would move the modifications out of the log statement, otherwise it's very hard to notice the actual changes done.

**Comment by aermakov-zalando** [comment]
👍

**Comment by jonathanbeber** [comment]
:+1:

**Pull request merged**